querybook/webapp/components/QueryEditor/QueryEditor.tsx

```diff
@@ -183,27 +188,47 @@ export class QueryEditor extends React.PureComponent<
         lineWrapping: boolean,
         readOnly: boolean,
         theme: string,
-        language: string,
-        metastoreId: number,
-        keyMap: CodeMirrorKeyMap,
-        lintingOn: boolean
+        getLintErrors: Nullable<
+            (
+                query: string,
+                editor: CodeMirror.Editor
+            ) => Promise<ILinterWarning[]>
+        >,
+        onLintCompletion: Nullable<(hasError?: boolean) => void>,
+        keyMap: CodeMirrorKeyMap
     ) {
-        const lintingOptions =
-            lintingOn && language != null && metastoreId != null
-                ? {
-                      lint: metastoreId
-                          ? {
-                                // Lint only when you can edit
-                                getAnnotations: getSqlLintAnnotations(
-                                    metastoreId,
-                                    language
-                                ),
-                                async: true,
-                                delay: 1000,
-                            }
-                          : null,
-                  }
-                : {};
+        const lintingOptions = getLintErrors
+            ? {
+                  lint: {
+                      // Lint only when you can edit
+                      getAnnotations: async (
```

Comment:
do you wanna have a separate function for it?  it seems too complicated as an inline one to me.